flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/RowDataToProtoSerializer.java

```diff
@@ -147,9 +150,8 @@ public Object toProtoValue(
                                 "Timestamp(millis)");
                     } else {
                         TimestampData timestampData = element.getTimestamp(fieldNumber, 6);
-                        long micros = getMicrosFromTsData(timestampData);
                         return AvroToProtoSerializer.AvroSchemaHandler.convertTimestamp(
-                                micros, true, "Timestamp(micros)");
+                                timestampData.getMillisecond(), true, "Timestamp(micros)");
```

Comment:
I don't understand this change. Why are we fetching millis from timestamp and treating it as micros?